tests/tests.h

```diff
@@ -591,6 +592,43 @@ struct MPILogInitAll
 };
 
 
+#ifdef DEAL_II_COMPILER_CUDA_AWARE
+// By default, all the ranks will try to access the device 0. We can distribute
+// the load better by using a random device which is done in this function. In
```

Comment:
What's the reason for that? Is the first graphics card always less occupied than the rest? Or are the other graphics cards just much slower?

Reply:
On the machines I am currently using I always have to fight for memory and chances are much higher to pass the tests if I try in subsequent testsuite runs different accessing different graphic cards.
Of course, I can also just use this setting locally if that is better for your setup.

Reply:
The two cards are the same. I think that we timeout because the kernel is never launched not because the computation does not finish. I get this problem sometimes on my desktop but I have no idea why this happens.

Reply:
> Of course, I can also just use this setting locally if that is better for your setup.

I would prefer the tester to be more reliable. Right now it's hard to know what's a real bug and what's a random failure.